controllers/volumereplicationgroup_controller.go

```diff
@@ -911,38 +919,48 @@ func (v *VRGInstance) processAsSecondary() (ctrl.Result, error) {
 
 	v.instance.Status.LastGroupSyncTime = nil
 
-	requeue := v.reconcileAsSecondary()
+	result := v.reconcileAsSecondary()
 
 	// If requeue is false, then VRG was successfully processed as Secondary.
 	// Hence the event to be generated is Success of type normal.
 	// Expectation is that, if something failed and requeue is true, then
 	// appropriate event might have been captured at the time of failure.
-	if !requeue {
+	if !result.Requeue {
 		rmnutil.ReportIfNotPresent(v.reconciler.eventRecorder, v.instance, corev1.EventTypeNormal,
 			rmnutil.EventReasonSecondarySuccess, "Secondary Success")
 	}
 
 	if err := v.updateVRGStatus(true); err != nil {
-		requeue = true
+		result.Requeue = true
 	}
 
-	if requeue {
-		v.log.Info("Requeuing resource as Secondary")
+	return result, nil
+}
 
-		return ctrl.Result{Requeue: true}, nil
-	}
+func (v *VRGInstance) reconcileAsSecondary() ctrl.Result {
+	vrg := v.instance
+	s3StoreAccessors := v.s3StoreAccessorsGet()
+	result := ctrl.Result{}
+	result.Requeue = v.reconcileVolSyncAsSecondary() || result.Requeue
```

Comment:
The change here is to progress with VR as secondary even if volsync as secondary is not complete, which is different from prior instance. This should not be an issue, as the PVC separation for VR/Volsync happens earlier, so this should be safe to call. Leaving a note here for @BenamarMk to take a look as well.